Error compiling on armv5tel architecture

Hi, totally a rust noob so I apologize if I'm misreading what the rust compiler is telling me.
I'm trying to build homeassistant on an embedded armv5tel NAS (32-bit). I'm stuck at building maturin, which requires indicatif v0.17.0. While building, I get the following error:

```
error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
 --> /home/homeassistant/.cargo/registry/src/github.com-1285ae84e5963aae/indicatif-0.17.0/src/state.rs:2:25
  |
2 | use std::sync::atomic::{AtomicU64, AtomicU8, Ordering};
  |                         ^^^^^^^^^
  |                         |
  |                         no `AtomicU64` in `sync::atomic`
  |                         help: a similar name exists in the module: `AtomicU8`

For more information about this error, try `rustc --explain E0432`.
error: failed to compile `maturin v0.13.2 (/home/homeassistant/maturin)`, intermediate artifacts can be found at `/home/homeassistant/maturin/target`

Caused by:
  could not compile `indicatif` due to previous error
```

Some info about my environment:

```
(homeassistant) homeassistant@nsa325:~/maturin$ rustc --version
rustc 1.59.0
(homeassistant) homeassistant@nsa325:~/maturin$ cargo --version
cargo 1.56.0
(homeassistant) homeassistant@nsa325:~/maturin$ uname -a
Linux nsa325 5.15.5-kirkwood-tld-1 #1.0 PREEMPT Sun Nov 28 23:08:13 PST 2021 armv5tel GNU/Linux
(homeassistant) homeassistant@nsa325:~/maturin$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

It seems like `AtomicU64` is not defined on all architectures (I've seen similar reports for PowerPC, i.e. https://github.com/tikv/rust-prometheus/issues/315). Can you please provide a fix/workaround?

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -12,6 +12,7 @@ jobs:
         os: [ubuntu-latest, macos-latest, windows-latest]
         rust: [stable, beta, 1.56]
         features: [--all-features, ""]
+        target: [""]
         exclude:
           - os: ubuntu-latest
             rust: beta
@@ -56,28 +57,48 @@ jobs:
           - os: ubuntu-latest
             rust: 1.56
             features: --features improved_unicode
+          - os: ubuntu-latest
+            rust: stable
+            features: --all-features
+            target: --target armv5te-unknown-linux-gnueabi
+            use-cross: true
 
     runs-on: ${{ matrix.os }}
 
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/github-script@v6
+        id: rust-target
+        with:
+          script: |
+            const target = '${{ matrix.target }}'
+            if (target) {
+              const rustTarget = target.slice('--target'.length).trim()
+              core.setOutput('rust-target', rustTarget)
+            } else {
+              core.setOutput('rust-target', '')
+            }
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: ${{ matrix.rust }}
+          target: ${{ steps.rust-target.outputs.rust-target }}
           override: true
       - uses: actions-rs/cargo@v1
         with:
+          use-cross: ${{ matrix.use-cross }}
           command: build
-          args: --workspace --all-targets
+          args: --workspace --all-targets ${{ matrix.target }}
       - uses: actions-rs/cargo@v1
         with:
+          use-cross: ${{ matrix.use-cross }}
           command: test
-          args: --workspace
+          args: --workspace ${{ matrix.target }}
       - uses: actions-rs/cargo@v1
         with:
+          use-cross: ${{ matrix.use-cross }}
           command: test
-          args: --workspace ${{ matrix.features }}
+          args: --workspace ${{ matrix.features }} ${{ matrix.target }}
 
   lint:
     runs-on: ubuntu-latest

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,6 +15,7 @@ rust-version = "1.56"
 [dependencies]
 console = { version = "0.15", default-features = false, features = ["ansi-parsing"] }
 number_prefix = "0.4"
+portable-atomic = "0.3.15"
 rayon = { version = "1.1", optional = true }
 tokio = { version = "1", optional = true, features = ["fs", "io-util"] }
 unicode-segmentation = { version = "1", optional = true }

--- a/src/progress_bar.rs
+++ b/src/progress_bar.rs
@@ -1,6 +1,6 @@
-use std::borrow::Cow;
 #[cfg(test)]
-use std::sync::atomic::{AtomicBool, Ordering};
+use portable_atomic::{AtomicBool, Ordering};
+use std::borrow::Cow;
 use std::sync::{Arc, Condvar, Mutex, MutexGuard, Weak};
 use std::time::{Duration, Instant};
 use std::{fmt, io, thread};

--- a/src/state.rs
+++ b/src/state.rs
@@ -1,8 +1,9 @@
 use std::borrow::Cow;
-use std::sync::atomic::{AtomicU64, AtomicU8, Ordering};
 use std::sync::Arc;
 use std::time::{Duration, Instant};
 use std::{fmt, io};
+
+use portable_atomic::{AtomicU64, AtomicU8, Ordering};
 
 use crate::draw_target::ProgressDrawTarget;
 use crate::style::ProgressStyle;

